src/test/java/com/fittracker/fittracker/service/WeightServiceTest.java

```diff
@@ -0,0 +1,100 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class WeightServiceTest {
+
+    @Mock
+    private WeightRepository weightRepository;
+
+    @InjectMocks
+    private WeightService weightService;
+
+    private final LocalDate TEST_DATE = LocalDate.of(2023,10,10);
+
+    private final Double TEST_VALUE = 100.1;
+
+    private final Weight weight = new Weight(TEST_DATE, TEST_VALUE);
+
+    private final WeightRequest weightRequest = new WeightRequest(TEST_DATE,TEST_VALUE);
+
+    private final WeightResponse weightResponse = new WeightResponse(TEST_DATE,TEST_VALUE);
+
+    @Nested
+    class FindByDate {
+        @Test
+        void givenDateFound_shouldReturnWeightResponse() {
+            when(weightRepository.findByDate(any())).thenReturn(Optional.of(weight));
+
+            var result = weightService.findByDate(LocalDate.of(2023, 10, 10));
+
+            assertThat(result).usingRecursiveComparison().isEqualTo(weightResponse);
+            verify(weightRepository).findByDate(LocalDate.of(2023, 10, 10));
+            verifyNoMoreInteractions(weightRepository);
+        }
+
+        @Test
+        void givenNoDateFound_shouldThrowWeightNotFoundException() {
+            when(weightRepository.findByDate(any())).thenReturn(empty());
+
+            assertThatThrownBy(() -> weightService.findByDate(LocalDate.of(2023, 10, 10)))
+                    .isInstanceOf(WeightNotFoundException.class)
+                    .hasMessageContaining("Weight not found for date: 2023-10-10" );
+
+            verify(weightRepository).findByDate(LocalDate.of(2023, 10, 10));
+            verifyNoMoreInteractions(weightRepository);
+        }
+
+    }
+
+    @Nested
+    class Save {
+        @Test
+        void givenNoDateFound_shouldSaveAndReturnWeightResponse() {
+            when(weightRepository.existsByDate(any())).thenReturn(false);
+            when(weightRepository.save(any())).thenReturn(weight);
+
+            var result = weightService.save(weightRequest);
+
+            assertThat(result).usingRecursiveComparison().isEqualTo(weightResponse);
+            verify(weightRepository).existsByDate(TEST_DATE);
+            verify(weightRepository).save(any());
+            verifyNoMoreInteractions(weightRepository);
+
+        }
+        @Test
+        void givenDateFound_shouldThrowException() {
+            when(weightRepository.existsByDate(any())).thenReturn(true);
+
+            assertThatThrownBy(()-> weightService.save(weightRequest))
+                    .isInstanceOf(WeightAlreadyExistsException.class)
+                    .hasMessageContaining("Weight already exists for date: 2023-10-10");
+            verify(weightRepository).existsByDate(any());
```

Comment:
we can verify in the same as above: `verify(weightRepository).existsByDate(TEST_DATE)`